x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -129,6 +129,7 @@ ResolvedIndices resolveIndicesAndAliases(String action, IndicesRequest indicesRe
             final boolean replaceWildcards = indicesOptions.expandWildcardsOpen() || indicesOptions.expandWildcardsClosed();
 
             // check for all and return list of authorized indices
+//            if (isAllIndices(indicesRequest.indices())) {
             if (IndexNameExpressionResolver.isAllIndices(indicesList(indicesRequest.indices()))) {
                 if (replaceWildcards) {
```

Comment:
I think you realized that the lone `*` index expression with `replaceWildcards == false` now behaves slightly different.
Before, Security would leave the `*` in-place and would leave core do its thing (fail in some way).
Now, we either fail in Security with index not found exception, or pass it to core as `*, -*`, which would also probably fail (again because request options say to not replace wildcards).
I'm not terribly concerned, though.

Reply:
No, I didn't realise that. Thanks for raising it. I was relying on the tests to catch any inconsistent behaviour and it seems we have some test gap.

I did some more digging and found there are behaviour differences:

1. `GET */_search?expand_wildcards=none&allow_no_indices=false` -> `404` Error
    * **Existing** -> **`404` Error**
    * **Proposed** -> `404` Error
2. `GET */_search?expand_wildcards=none&allow_no_indices=true` -> 
    * **Existing** -> **Empty search result**
    * **Proposed** -> `404` Error

The difference is about how `allow_no_indices` is intepreted. I'd argue that the existing behaviour is not accurate because `allow_no_indices` implies "expansion" of either wildcards or aliases. But the request explicilty asks for no expansion. Hence the name `*` should be treated as a literal index name. In that case, the relevant option should be `ignore_unavailable` instead of `allow_no_indices`. That is the following request should return empty result

3. `GET */_search?expand_wildcards=none&ignore_unavailable=true` 
    * **Existing** -> Empty result
    * **Proposed** -> Empty result

Based on the following comment, it seems that core is not fully convinced with its own behaviour either. 
https://github.com/elastic/elasticsearch/blob/eeb09f0248765c7b24a4ba8129e29141e1e7c956/server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java#L218-L223

That said, behaviour change is what I tried to avoid and it is the reason why I bothered to add special handling for error message. I'll make some adjustments. Thanks again for noticing it.